orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -165,6 +178,206 @@ def save_as_onnx(self, path):
+            set([n.name for n in onnx_model.graph.initializer])), \
+            "Initializer names do not match between PyTorch model and ONNX model, " \
+            "please report a bug to ONNX Runtime."
+
+        return onnx_model
+
+    def _init_onnx_model(self, inputs):
+            if self._onnx_model is not None:
+                return
+
+            if self._torch_model is not None:
+                # NOTE: pt model is moved to cpu to conserve gpu memory.
+                self._torch_model.cpu()
+                # torch buffers created using 'register_buffer' are not meant to be trainable.
+                torch_buffers = list(dict(self._torch_model.named_buffers()).keys())
+                #self.frozen_weights_ = self.frozen_weights_ + torch_buffers
+                self._onnx_model = self.convert_torch_model_loss_fn_to_onnx(torch.device('cpu'), inputs)
+
+            self._init_session()
+
+    def _prepare_input_and_fetches(self, input_desc_with_, internal_learning_rate, internal_loss_scale, *args, **kwargs):
+            fetches = None
+            """
+            if type(args) == tuple and len(args) == 1 and type(args[0]) == list:
+                input = tuple(args[0])
+            else:
+                input = args
+            """
+            input = args #added
+
+            for input_desc in input_desc_with_:
+                if input_desc[0] in kwargs:
+                    input = input + (kwargs[input_desc[0]],)
+            if internal_learning_rate is not None:
+                input = input + (internal_learning_rate,)
+            if internal_loss_scale is not None:
+                input = input + (internal_loss_scale,)
+            """
+            elif self.use_mixed_precision:
+                # loss_scale input name is needed to call train_step, for example:
+                #   kwargs[model.loss_scale_input_name] = loss_scale
+                #   outputs = model.train_step(*args, **kwargs)
+                # However, when first time train_step is called model.loss_scale_input_name is not set.
+                # To workaround this problem, we use the special name 'default_loss_scale_input_name' to indicate
+                # the loss_scale.
+                if 'default_loss_scale_input_name' in kwargs.keys():
+                    input = input + (kwargs['default_loss_scale_input_name'],)
+            """
+            fetches = None
+            if 'fetches' in kwargs:
+                fetches = kwargs['fetches']
+
+            return input, fetches
+
+    def _init_session(self):
+            if self._onnx_model is None:
+                return
+
+            if self._enable_internal_postprocess:
+                self._onnx_model = postprocess.run_postprocess(self._onnx_model)
+
+            if self._extra_postprocess:
+                self._extra_postprocess(self._onnx_model)
+            return
+
+    # temp function for testing
+    def get_onnx(self):
```

Comment:
don  forget to remove it in the final version. we can always access it directly by accessing _oonx_model